### ISSUE TITLE
[8.0.0-beta-3 - Nuxt 3 RC13] Route for page does not include locale

### ISSUE THREAD

**Issue by ineshbose** [nuxt-modules/i18n#1620]
### Version
@nuxtjs/i18n: 8.0.0-beta.3
nuxt: 3.0.0-rc.12

### @nuxtjs/i18n configuration

```
i18n: {
  lazy: true,
  langDir: 'locales',
  strategy: 'prefix',
  defaultLocale: 'en',
  detectBrowserLanguage: {
    useCookie: true,
    cookieKey: 'i18n_redirected',
    redirectOn: 'root',
  },
  locales: [
    {
      code: 'en',
      name: 'English',
      file: 'en.json',
      customFunctions: {
        greeting({ name }) {
           return `Hello ${name}.`
        },
      },
    },
    {
      code: 'es',
      name: 'Español',
      file: 'es.json',
    },
  ],
  vueI18n: {
    legacy: false,
    locale: 'en',
  },
}
```

### Reproduction Link

Have a page with the sample code:

```vue
<script setup>
defineI18nRoute({
  paths: {
    en: '/about',
    es: '/sobre',
  },
});

definePageMeta({
  layout: false,
});
// ...
</script>
```

### What is actually happening?

https://github.com/nuxt-community/i18n-module/blob/next/src/pages.ts#L167 picks up on `false` in the later part of the code and hence the route isn't included in localisation.

**Comment by ineshbose** [comment]
Suggested solution: The condition can be moved to AFTER determining `result` and test if for that at https://github.com/nuxt-community/i18n-module/blob/next/src/pages.ts#L182

```js
  if (REGEX_FALSE.test(result)) {
    return 'false'
  }
```

But personally I don't see a reason for this code though, because if `defineI18nRoute(false)` were to be used, `result` would be `false` (and hence returned) regardless, while line endings or other characters will not affect `return (${macroContent})`.

I'll create a PR if you can advice how to proceed please @kazupon

**Comment by kazupon** [comment]
Thank you for your reporting!
I've faced this issue too.

We need to resolve with AST using the parser like `acorn` and the AST walker like `estree`.
Currently, We’re extracting with this module parser and regex.
this way cannot recognize javascript syntax-semantics.
I’ll try to use the fix soon.

**Comment by ineshbose** [comment]
I agree that resolving with ASTs is the way to go instead of regex!
Meanwhile I've edited the files in `node_modules` for it to work for my application. 🙂

**Comment by kazupon** [comment]
Hi!
I've just fixed #1636 
You can try to use the edge version! :)
